sequencing/ATACSeq/jobSubmission/3_batchRunPeakCalling.sh

```diff
@@ -6,107 +6,158 @@
-echo "Changing Folder to: "
-echo $SLURM_SUBMIT_DIR
 
-cd $SLURM_SUBMIT_DIR
-
-## load config file provided on command line when submitting job
+## load config file provided on command line related to the specified project
+source $1
 echo "Loading config file for project: " $1
-export PROJECT=$1
-
-source ./ATACSeq/config/config.txt 
 echo "Project directory is: " $DATADIR
 
+## Log files directory
+LOG_DIR=ATACSeq/logFiles/${USER}/${SLURM_ARRAY_JOB_ID}
+echo "Log files will be moved to dir: " $LOG_DIR
+mkdir -p $LOG_DIR
+mv ATACPeakCallingS3-${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}* $LOG_DIR
+
 ##check array specified and exit if not
 if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
 then 
     { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
 fi
 
-#-----------------------------------------------------------------------#
-
-echo "Changing Folder to Data directory "
-echo ${ALIGNEDDIR}
+## check step method matches required options and exit if not
+if [[ ! $2 =~ "SHIFT" ]] && [[ ! $2 =~ "PEAKS" ]] && [[ ! $2 =~ "HMMRATAC" ]] && [[ ! $2 =~ "FRIP" ]] &&[[ ! $2 == '' ]];
+then 
+    { echo "Unknown step specified. Please use SHIFT, PEAKS, HMMRATAC, FRIP or some combination of this as a single string (i.e. FASTQC,TRIM)" ; exit 1; }            
+fi
 
 cd ${ALIGNEDDIR}
 BAMFILES=($(ls *.filt.nodup.bam))
-
-echo "Number of bam files found for alignment:"" ""${#BAMFILES[@]}"""	
-
-sample=${BAMFILES[${SLURM_ARRAY_TASK_ID}]}
-sampleName=$(basename ${sample%.filt.nodup.bam})
-
-
+echo "Number of aligned bam files: "" ""${#BAMFILES[@]}"""
+  
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## If only $1 is specified, all steps will be run
+## option SHIFT: shift aligned reads for single-end peak calling   
 if [ $# = 1 ] || [[ $2 =~ 'SHIFT' ]]
 then
 	module load BEDTools
 	module load SAMtools
 	module load R/3.6.3-foss-2020a
-	## shift reads for peak calling
-	echo "Shifting reads"
-
-	cd ${SCRIPTDIR}
-	./ATACSeq/preprocessing/shiftAlignedReads.sh ${sampleName}
-
-	if [[ $? == 0 ]]
-		then date -u
-		echo "Reads shifted"
-	fi
+ 
+	echo "Step 3.1 SHIFT started. Aligned reads will be shifted"
+  mapfile -t SAMPLES < ${METADIR}/samples.txt
+  sample=${SAMPLES[${SLURM_ARRAY_TASK_ID}]}
+  echo "Sample(s) specified by array number in command line is/are: " ${sample}
+  
+  if [[ -s ${ALIGNEDDIR}/${sample}.filt.nodup.bam ]]
+  then 
+    cd ${SCRIPTDIR}
+	  sh ./ATACSeq/preprocessing/shiftAlignedReads.sh ${sample}
+  else
+    echo "Aligned bam file for ${sample} not found. Aligned reads can't be shifted."
+  fi
 fi
 
+## option PEAKS: peak calling is performed in two ways using MACS3: single-end and paired-end 
 if [ $# = 1 ] || [[ $2 =~ 'PEAKS' ]]
 then
-	echo Starting peak calling at:
-	date -u
+
 	module purge
-	module load MACS2/2.1.2.1-foss-2017b-Python-2.7.14
+	module load Python/3.9.6-GCCcore-11.2.0-bare
+  source /lustre/home/${USER}/pythonEnv/bin/activate
 	module load BEDTools
```

Comment:
These also are not compatible anymore. Python is using GCC version 11.2.0, which we don't have a version of BEDtools for currently.

You could send in a ticket asking for some compatibility if you require python 3.9.6 specifically. If not then:
Python/3.8.2-GCCcore-9.3.0

will work in place of the current version of python